pkg/glusterfs/controllerserver.go

```diff
@@ -59,11 +60,27 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
 	var gdReq api.VolCreateReq
 
 	reqConf.gdVolReq = &gdReq
-
+	snapshotEnabled := true
 	// Get Volume name
 	if req != nil {
 		reqConf.gdVolReq.Name = req.Name
 	}
+
+	for k, v := range req.GetParameters() {
+
+		switch strings.ToLower(k) {
+
+		// Placeholder for more SC parameters.
+		case "snapshotenabled":
```

Comment:
Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request